### PR TITLE
Move to Python

### PR DESCRIPTION
**Pull request by SunilMohanAdapa**
We can run Freedom Maker like a tool on the command line and package it for Debian.

- Remove Makefile and shell script used for vmdebootstrap invocation and
  prefer a python tool instead of Makefile and shell scripts.

- Object oriented behavior for image builders improving code reuse.

- Thorough tests to test invocation of various tests and invocation of
  tool itself.

- Exception handling for proper cleanups.

- Revised logging behavior, somewhat incomplete.

I have tested this for amd64, virtualbox-i386 and all current test cases pass.  Will perform full testing in time for next release where we build images: 0.9

### CODE REVIEW

File: freedommaker/tests/vmdebootstrap-stub [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/python3
+#
+# This file is part of Plinth.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
```
Comment (jvalleroy) [inline freedommaker/tests/vmdebootstrap-stub]: Should this be GPL, not AGPL?
(same with setup.py)

Comment (SunilMohanAdapa) [inline freedommaker/tests/vmdebootstrap-stub]: This is a simple copy/paste error.  I have modified my commit and pushed the newer version. I have also added another commit updating the README.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python3
+#
+# This file is part of Plinth.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
```
Comment (jvalleroy) [inline setup.py]: Please change this one too :)

Comment (SunilMohanAdapa) [inline setup.py]: Ouch! Did a grep on older commits.  Fixed now. :)  Does the rest look okay?

**Comment by jvalleroy** [comment]
Looks good, merged.

**Pull request merged**